added the method requireGrammarsSync() for loading grammars from npm modules

bcoe opened the pull request:
I've added the method `requireGrammarsSync()`, for loading a grammar into the registry from an npm module. This is super useful when using highlights as a stand-alone library, and allows grammars to be published to npm for just this purpose.

FYI, we recently switched to highlights as our syntax highlighter for npm. awesome work!

![screen shot 2015-01-30 at 12 15 20 am](https://cloud.githubusercontent.com/assets/194609/5972896/d15afada-a815-11e4-97a3-c6f3956424f3.png)

kevinsawicki commented on src/highlights.coffee:
```diff
@@ -30,6 +31,23 @@ class Highlights
+    @loadGrammarsSync()
+
+    packageDir = path.dirname(require.resolve(moduleName))
+    grammarsDir = path.resolve(packageDir, './grammars')
```
Is the `./` needed here? Won't it still be relative to `packageDir` without it?

kevinsawicki commented on package.json:
```diff
@@ -22,15 +22,16 @@
     "underscore-plus": "^1.5.1",
-    "fs-plus": "^2.2.6"
+    "fs-plus": "^2.2.6",
+    "season": "^5.0.2"
```
Would you mind bumping this to `^5.1.2`, there were some recent bug fixes that would be nice to pull in.

kevinsawicki commented on src/highlights.coffee:
```diff
@@ -30,6 +31,23 @@ class Highlights
+  # Public: allows a grammar to be loaded from
+  #   an npm module.
+  #  :moduleName - the module to require.
+  requireGrammarSync: ({moduleName}={}) ->
```
This should probably be `requireGrammarsSync` (plural) since it loads all the grammars from the module.

bcoe replied:
Good call, naming things is hard!

bcoe commented:
@kevinsawicki pull request updated, thanks for the quick code-review I've giving a demo of highlights to everyone at npm today this will add a lot of razzmatazz.

bcoe commented:
@kevinsawicki hold off on merging this, I found something that needs to be patched related to where modules are included from.

bcoe commented:
The problem is that `require.resolve()` always resolves to highlights' `node_modules` folder, perhaps instead use `path.resolve(__dirname, '../', 'node_modules', moduleName). I'll ask @isaacs if there's a better approach.

bcoe commented:
Based on a in-the-air conversation with @isaacs regarding trusting `__dirname`, i.e., you should not, I think this is a good fix:

```javascript
highlighter.requireGrammarsSync({
  modulePath: require.resolve('atom-language-dart')
})
```
I'd say this is ready to go.

kevinsawicki commented on src/highlights.coffee:
```diff
@@ -30,6 +31,23 @@ class Highlights
+    for file in fs.readdirSync(grammarsDir)
+      grammarPath = path.join(grammarsDir, file)
+      continue unless CSON.resolve(grammarPath)
+      grammar = @registry.createGrammar(grammarPath, CSON.readFileSync(grammarPath))
```
This should probably use `@registry.loadGrammarSync` since it reads it, adds it, and validates it as well.

So something like:

```coffee
for file in fs.readdirSync(grammarsDir)
  if grammarPath = CSON.resolve(path.join(grammarsDir, file))
    @registry.loadGrammarSync(grammarPath)
```

bcoe replied:
Cool, will update after lunch. The `require.resolve()` seems reasonable otherwise.

bcoe commented:
@kevinsawicki okay I think this is actually close to ready to go now, to summarize:

* adds the method `requireGrammarsSync()`, which based on the advice of my coworkers now recommends using:

```javascript
highlighter.requireGrammarsSync({
  modulePath: require.resolve('atom-language-dart/package.json')
})
```

The nice thing about this is that the existing atom language modules have package.json files, so existing grammars can be published to npm with no modifications.

* I've switched to using `@registry.loadGrammarSync` based on your suggestion.
* I've bumped `season` to version to `v5.1.2` based on your request, I've also moved `season` into `dependencies` from `devDependencies`, as it is required by `requireGrammarsSync`.

Thanks for your patience.

kevinsawicki commented:
Thanks for this, published in `1.1.0`
